sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/OptimizeSkewedJoin.scala

```diff
@@ -55,6 +55,11 @@ import org.apache.spark.sql.internal.SQLConf
  */
 object OptimizeSkewedJoin extends Rule[SparkPlan] {
 
+  /**
+   * The list of [[ShuffleOrigin]]s supported by this rule.
+   */
+  val supportedShuffleOrigins: Seq[ShuffleOrigin] = Seq(ENSURE_REQUIREMENTS)
```

Comment:
should the rule `OptimizeSkewedJoin` itself check it?

Reply:
Yes, we better do, to make it more robust, although it's not really necessary here as we exclude these rules if there's been a repartition shuffle.